src/app/messagefilter.cpp

```diff
@@ -50,6 +48,7 @@ QObject* MessageFilter::source() const
 void MessageFilter::setSource(QObject* source)
 {
     QSortFilterProxyModel::setSourceModel(qobject_cast<QAbstractItemModel*>(source));
+    m_connection = static_cast<const MessageModel*>(sourceModel())->buffer()->connection();
```

Comment:
can you store the buffer instead, so we can use it for both things in filterAcceptsRow():
- m_buffer->isChannel() for limiting to channels
- m_buffer->connection() for accessing the connection count